### PR TITLE
Replace jQuery shorthands with their full counterparts

### LINKED ISSUE #1390: Stop using jQuery shorthands
Several shorthands for jQuery [are scheduled for deprecation](https://github.com/jquery/jquery/issues/4796#issuecomment-702022117) ([another source](https://github.com/jquery/api.jquery.com/issues/972)) but haven't been documented yet.

We're using these shorthands in several locations throughout the JS, and we should stop using them in favor of the full version.

Examples:

- `.click()` to `.trigger( 'click' )`.
- `.blur()` to `.trigger( 'blur' )`.

### PR DESCRIPTION
**Pull request by reykjalin**
Make sure #1392 is merged before this PR to bring the composer v2 updates to `master`.

# Changes proposed in this Pull Request:

Fixes #1390

This removes all calls to the jQuery shorthands we've used in our code. I located the shorthands in use with the following regex, as described in [this comment](https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1390#issuecomment-738917808):

```regex
\.(blur|focus|focusin|focusout|resize|scroll|click|dblclick|mousedown|mouseup|mousemove|mouseover|mouseout|mouseenter|mouseleave|change|select|submit|keydown|keypress|keyup|contextmenu)\(
```

# Testing instructions

- Make sure the `.on( 'change' )` handlers are working as intended for the Payment Request buttons. I'm honestly not sure how to properly test this, but I _think_ it's just making sure that they render with the correct brand?
- Make sure that the credit card input loses focus when an error occurs during checkout (the `blur` action).
- I'm not sure what the `submit` trigger is doing in `sourceResponse`, but I assume there's some way to test this?

Here's the `sourceResponse` function, for a quick glance:

```js
		sourceResponse: function( response ) {
			if ( response.error ) {
				return $( document.body ).trigger( 'stripeError', response );
			}
			wc_stripe_form.reset();
			wc_stripe_form.form.append(
				$( '<input type="hidden" />' )
					.addClass( 'stripe-source' )
					.attr( 'name', 'stripe_source' )
					.val( response.source.id )
			);
			if ( $( 'form#add_payment_method' ).length ) {
				$( wc_stripe_form.form ).off( 'submit', wc_stripe_form.form.onSubmit );
			}

			wc_stripe_form.form.submit();
			wc_stripe_form.form.trigger( 'submit' );
		},
```

-------------------
- [x] Make sure your changes respect [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- [x] Did you make changes, or create a **new .js file**? If **Gruntfile.js** exists in the repo, make sure to run `grunt`.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     },
     "require-dev": {
 				"phpunit/phpunit": "^4.8 || ^5.0",
-				"dealerdirect/phpcodesniffer-composer-installer": "^0.5.0",
+				"dealerdirect/phpcodesniffer-composer-installer": "^0.7.0",
```
Comment (v18) [inline composer.json:13]: I'm curious about this line - is this a required change for this PR to work?

Comment (reykjalin) [reply composer.json:13]: From the PR description:

>Make sure #1392 is merged before this PR to bring the composer v2 updates to master.

In hindsight I should've just not commited this, sorry about that.

We can safely drop the composer changes out of the PR, but alternatively we could merge #1392 and this diff will go away.

File: assets/js/stripe.js [hunk 2]
```diff
@@ -528,7 +528,7 @@ jQuery( function( $ ) {
 				$( wc_stripe_form.form ).off( 'submit', wc_stripe_form.form.onSubmit );
 			}
 
-			wc_stripe_form.form.submit();
+			wc_stripe_form.form.trigger( 'submit' );
```
Comment (allendav) [inline assets/js/stripe.js:531]: Are you sure submit is being deprecated? I couldn't find that in the linked docs, nor on https://api.jquery.com/submit/

**Review by allendav — CHANGES_REQUESTED** [review]
Tests fine. One question about submit. If that isn't really being deprecated, we shouldn't change that one.

File: assets/js/stripe.js (hunk 2 above)
Comment (reykjalin) [reply assets/js/stripe.js:531]: None of these deprecations are documented. I used the list [here](https://github.com/jquery/jquery/blob/305f193aa57014dc7d8fa0739a3fefd47166cd44/src/event/alias.js#L10-L21) to figure out which shorthands are being deprecated.

There are more details on how I found the shorthands we use in [this comment](https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1390#issuecomment-738917808) 🙂 

For information about the deprecations, see [this jQuery issue](https://github.com/jquery/api.jquery.com/issues/972).

**Comment by reykjalin** [comment]
@allendav I've replied to the comment. I think we're good since none of these deprecations are documented yet, but these shorthands will be deprecated eventually. I still put the `needs review` label on the PR again instead of merging directly, so it'd be great if you could take a look 🙂

**Comment by allendav** [comment]
@reykjalin - ah - I see - ok, let's proceed

**Review by allendav — APPROVED** [review]

**Pull request merged**